plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/loops/routing/PolylineSelfLoopRouter.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Kiel University and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.elk.alg.layered.intermediate.loops.routing;
+
+import java.util.Iterator;
+
+import org.eclipse.elk.alg.layered.graph.LPort;
+import org.eclipse.elk.alg.layered.intermediate.loops.SelfLoopEdge;
+import org.eclipse.elk.core.math.KVector;
+import org.eclipse.elk.core.math.KVectorChain;
+
+import com.google.common.math.DoubleMath;
+
+/**
+ * Routes self loops with polylines. This is basically the same routing as computed by the
+ * {@link OrthogonalSelfLoopRouter}, but with the corners cut a little.
+ */
+public class PolylineSelfLoopRouter extends OrthogonalSelfLoopRouter {
+    
+    /** Distance by which corner bend points are offset to cut their corner. */
+    private static final double CORNER_DISTANCE = 10;
+
+    /**
+     * Turns a vector chain of orthogonal bend points into polyline bend points by cutting the corners.
+     */
+    @Override
+    protected KVectorChain modifyBendPoints(final SelfLoopEdge slEdge, final EdgeRoutingDirection routingDirection,
+            final KVectorChain bendPoints) {
+        
+        // Add the source and target points
+        LPort lSourcePort = slEdge.getSLSource().getLPort();
+        bendPoints.add(0, lSourcePort.getPosition().clone().add(lSourcePort.getAnchor()));
+
+        LPort lTargetPort = slEdge.getSLTarget().getLPort();
+        bendPoints.add(lTargetPort.getPosition().clone().add(lTargetPort.getAnchor()));
+        
+        return cornerfy(bendPoints, CORNER_DISTANCE);
+    }
+    
+    /** Tolerance for double comparisons. */
+    private static final double TOLERANCE = 0.01;
+    
+    /**
+     * Turns the given list of bend points into a new list of bend points by cutting the corners. The first and last
+     * bend point always remain untouched (and are not included in the returned list of bend points anyway). The other
+     * bend points are replaced by two each which are ideally the given distance away from the original bend point.
+     */
+    protected KVectorChain cornerfy(final KVectorChain bendPoints, final double distance) {
```

Comment:
`cornerify`?

Reply:
I'm not at all sure, so I renamed the method to `cutCorners(...)`. :D